src/update-build-cache.sh

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env bash
+
+set -eu
+
+if [[ $(uname -o) = 'Cygwin' ]]; then
+  GITHUB_WORKSPACE=$(cygpath "${GITHUB_WORKSPACE}")
+  WINDOWS=1
+else
+  WINDOWS=0
+fi
+
+if [[ ! -e ${GITHUB_WORKSPACE}/.github/caches/$1/cache.tar ]]; then
+  opam clean --logs --download-cache --all-switches --yes
+  # XXX Upstream should no longer install these by default
+  rm -f ~/.opam/build/bin/ocaml*.byte
+  if ((WINDOWS)); then
```

Comment:
The installation of various tools by default copies, rather than symlinking, so we end up with two copies of `ocamlc.opt.exe` and `ocamlc.exe`.